tough/src/cache.rs

```diff
@@ -96,7 +96,8 @@ impl Repository {
     {
         self.cache_file_from_transport(
             self.snapshot_filename().as_str(),
-            self.max_snapshot_size()?,
+            self.max_snapshot_size()?
+                .unwrap_or(self.limits.max_timestamp_size),
```

Comment:
I would prefer adding a separate `max_snapshot_size` limit to the limits struct. It can share the default with `max_timestamp_size` (1 MiB) but it's a different metadata file and clients may want to treat it differently.

Reply:
Yes, good point. I've fixed that with 44740517cbe5a74421b818682f8bb4fff637aa50

I will squash the commit once the review process is done